From: Kyungmin Park <kyungmin.park@samsung.com>
Date: Fri, 10 Aug 2007 21:00:21 +0000 (-0700)
Subject: [MIPS] i8259: Add disable method.
X-Git-Tag: linux-2.6.23-rc4~1
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=b445dddb98143ffa52e3e3d8b082f04b04f3e18b;hp=11d9842e7844b97f45b73a26b0982face31a3728

[MIPS] i8259: Add disable method.

After 76d2160147f43f982dfe881404cfde9fd0a9da21, the qemu NE2000 was
frequently producing WATCHDOG timeouts.

Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Cc: Ralf Baechle <ralf@linux-mips.org>
Cc: Ingo Molnar <mingo@elte.hu>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
---

diff --git a/arch/mips/kernel/i8259.c b/arch/mips/kernel/i8259.c
index 2345160..b6c3080 100644
--- a/arch/mips/kernel/i8259.c
+++ b/arch/mips/kernel/i8259.c
@@ -36,6 +36,7 @@ void mask_and_ack_8259A(unsigned int);
 static struct irq_chip i8259A_chip = {
 	.name		= "XT-PIC",
 	.mask		= disable_8259A_irq,
+	.disable	= disable_8259A_irq,
 	.unmask		= enable_8259A_irq,
 	.mask_ack	= mask_and_ack_8259A,
 };
